internal/repos/github.go

```diff
@@ -116,6 +128,31 @@ func newGithubSource(svc *types.ExternalService, c *schema.GitHubConnection, cf
 
 	token := &auth.OAuthBearerToken{Token: c.Token}
 
+	var (
+		v3Client     = github.NewV3Client(apiURL, token, cli)
+		v4Client     = github.NewV4Client(apiURL, token, cli)
+		searchClient = github.NewV3SearchClient(apiURL, token, cli)
+	)
+
+	dotcomMode := envvar.SourcegraphDotComMode()
+	if !dotcomMode || (dotcomMode && c.CloudGlobal) {
+		setCollector := func(resource string, getter func() *ratelimit.Monitor) {
+			rlm := getter()
+			if rlm == nil {
+				return
+			}
+			rlm.SetCollector(func(known bool, remaining float64) {
+				if !known {
+					return
+				}
+				githubRemainingGauge.WithLabelValues(resource, svc.DisplayName).Set(remaining)
+			})
+		}
+		setCollector("core", v3Client.RateLimitMonitor)
+		setCollector("graphql", v4Client.RateLimitMonitor)
+		setCollector("search", searchClient.RateLimitMonitor)
```

Comment:
Perhaps a bit simpler?

```suggestion
		for resource, m := range map[string]*ratelimit.Monitor{
		    "core": v3client.RateLimitMonitor(),
		    "graphql": v4Client.RateLimitMonitor(),
		    "search": searchClient.RateLimitMonitor(),
		} {
		    m.SetCollector(func(remaining float64) {
				githubRemainingGauge.WithLabelValues(resource, svc.DisplayName).Set(remaining)
			})
		}
```

Reply:
Actually, I think you can do this.

```go
collector := githubRemainingGauge.WithLabelValues(resource, svc.DisplayName).Set
m.SetCollector(collector)
```